### ISSUE TITLE
Failed to create wallet from private key: Odd number of digits. 

### ISSUE THREAD

**Issue by aiwowo1** [foundry-rs/foundry#2086]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (9b2d95d 2022-06-23T00:13:03.193579407Z)

### What command(s) is the bug in?

forge script

### Operating System

Linux

### Describe the bug

I got this error 
```
Error:
   0: Failed to create wallet from private key: Odd number of digits
```
when running this command:
```
forge script script/MyContract.s.sol:MyScript --rpc-url $RINKEBY_RPC_URL --broadcast --verify --etherscan-api-key $ETHERSCAN_KEY -vvvv --private-key $PRIVATE_KEY
```
I located the problem is on the $PRIVATE_KEY. however, it can be successfully executed when I replace the $PRIVATE_KEY with the actual private key. And I have checked the key it self is not Odd digit. 

I also check the other two env variables don't have this problem.

**Comment by onbjerg** [comment]
Is the `PRIVATE_KEY` environment variable set?

**Comment by aiwowo1** [comment]
Yes. the PRIVATE_KEY is set in the .env along with other two. I can echo out all 3 variables.

**Comment by aiwowo1** [comment]
I run linux in WSL that might be a problem?

**Comment by aiwowo1** [comment]
the scirpt is like this in a bash file:

```
if [ -f .env ]
then
  export $(cat .env | xargs)
fi
echo $RINKEBY_RPC_URL
echo $RINKEBY_PRIV_KEY
forge script script/UniswapV2Pair.s.sol:MyScript --rpc-url $RINKEBY_RPC_URL --private-key $RINKEBY_PRIV_KEY
```

==$ bash script/run.sh==
```
https://rinkeby.infura.io/v3/xxxxxxxxxxxxxxxxxxxxxxxxxx  (masked)
1ff408e9e5342c7e3fa40bba9345da9b0fdaabf7a6e4eaceb7de1d6a1ba55391 (faked)
Error:
Failed to create wallet from private key: Odd number of digits
script/run.sh: line 12: $'\r': command not found
```

**Comment by onbjerg** [comment]
That's weird. My assumption would be that maybe some invisible character is added from your `.env`? My best guess is the `\r` that Windows adds. Try `echo $PRIVATE_KEY | cat -v` to see if that's the case

**Comment by aiwowo1** [comment]
That's exactly the problem.  There is a ^M at the end. I used to use ``` dos2unix filename ``` to reformat a bash file and this time I misread the message and didn't think of to reformat the .env file.  It worked now. Thanks for the help.

**Comment by onbjerg** [comment]
Added a fix in #2099 so it's not necessary to do it manually :)

**Comment by wuchuank** [comment]
Not sure if its just me but I'm still facing this error in the latest version of `forge 0.2.0 (114413c 2022-07-19T00:08:18.125698929Z)`

Able to bypass it by running dos2unix .env but it looks like the fix doesnt work? @onbjerg
